### PR TITLE
feat: add yaai-llm crate — LlmClient abstraction with OpenAI, Claude, and Stub backends

### PR DESCRIPTION
**Pull request by robdefeo**
## Summary

Introduces the `yaai-llm` crate, which provides a thin, provider-agnostic interface for sending a prompt to an LLM and receiving a response.

### `LlmClient` trait (`src/lib.rs`)
```rust
#[async_trait]
pub trait LlmClient: Send + Sync {
    async fn complete(&self, system: Option<&str>, messages: &[Message]) -> Result<LlmResponse>;
}
```
The single method accepts an optional system prompt and a slice of conversation `Message`s and returns an `LlmResponse`.

### Core types
| Type | Purpose |
|------|---------|
| `Message` | Typed conversation turn with `role` + `content`; constructors `::system`, `::user`, `::assistant` |
| `LlmResponse` | Carries optional `content: String` (text answer) and/or `tool_call: ToolCall` |
| `ToolCall` | Name + JSON arguments of a requested tool invocation |

`LlmResponse::is_final_answer()` returns `true` when the response contains text and no tool call — used by the agent loop to detect termination.

### Backends
- **`OpenAiClient`** (`src/openai.rs`) — POST `/v1/chat/completions`; injects the system prompt as the first message entry as required by the OpenAI API.
- **`ClaudeClient`** (`src/claude.rs`) — Anthropic Messages API (`/v1/messages`); passes the system prompt via the dedicated top-level `system` field; defaults to `claude-sonnet-4-5` when no model is specified.
- **`StubClient`** (`src/stub.rs`) — returns a pre-scripted sequence of `LlmResponse`s in order; errors with a clear message if the script is exhausted. Intended for all agent-level tests so no real API calls are made.

### Tests (`tests/stub_tests.rs`)
16 tests covering:
- `StubClient` ordering and exhaustion behaviour
- `is_final_answer` edge cases
- `Message` and `LlmResponse` serde round-trips
- `Box<dyn LlmClient>` delegation
- `ClaudeClient` / `OpenAiClient` construction

### CODE REVIEW

File: crates/llm/src/openai.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+impl OpenAiClient {
+    pub fn new(api_key: impl Into<String>, model: impl Into<String>) -> Self {
+        Self {
+            api_key: api_key.into(),
+            model: model.into(),
+            client: reqwest::Client::new(),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (robdefeo) [reply crates/llm/src/openai.rs:34]: https://github.com/robdefeo/yaai/pull/1/commits/6f8908c0010aa6cbcbd98426917ed4c580c14731

File: crates/llm/src/claude.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            DEFAULT_MODEL.to_string()
+        } else {
+            model
+        };
+        Self {
+            api_key: api_key.into(),
+            model,
+            client: reqwest::Client::new(),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (robdefeo) [reply crates/llm/src/claude.rs:46]: https://github.com/robdefeo/yaai/pull/1/commits/6f8908c0010aa6cbcbd98426917ed4c580c14731

File: crates/llm/src/stub.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    }
+}
+
+#[async_trait]
+impl LlmClient for StubClient {
+    async fn complete(&self, _system: Option<&str>, _messages: &[Message]) -> Result<LlmResponse> {
+        self.script.lock().unwrap().pop().ok_or_else(|| {
+            anyhow!("StubClient script exhausted — agent ran more steps than expected")
+        })
```
[the remark this answers is not in the document]
Comment (robdefeo) [reply crates/llm/src/stub.rs]: https://github.com/robdefeo/yaai/pull/1/changes/25ef5840b2f762d55a7e9b0d2f55dec98904a36f
